### PR TITLE
Tekton go unit test task

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Changes proposed in this pull request:

Add unit test task for Go code This pr adds a new task for running unit tests for Go code. The task is tailored to be triggered by prow as a prowjob and uses the golangci-lint tool to run the tests. The task is compatible with Tekton v0.36.0 and above and can be installed using the provided kubectl command. The task requires a workspace named "repo" where the sources to test are stored. The task has optional parameters for specifying the path to the Go code to test, as well as the GitHub org and repo that triggered the prowjob.  GitHub org and repo params values should be provided by prow. The commit also includes a sample presubmit prowjob configuration for running the unit tests.

**Related issue(s)**
See #6798

### CODE REVIEW

File: task/go-unit-tests/0.1/go-unit-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    tekton.dev/tags: test
+    tekton.dev/displayName: "unit-tests-go"
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: >-
+    A unit-test-go Task is used to run unit tests for go code.
+    Task is tailored to be triggered by prow as a prowjob.
+  workspaces:
+    - name: repo
+      description: "The workspace where sources to test are stored. Usually this it should be a shared workspace with other tasks."
```
Comment (IwonaLanger) [inline task/go-unit-tests/0.1/go-unit-tests.yaml:19]:
```suggestion
      description: "The workspace where tested sources are stored. Usually, this should be a workspace shared with other tasks."
```

File: task/go-unit-tests/0.1/go-unit-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - name: path-to-test
+      description: "The path to go code to test."
+      type: string
+      default: "./..."
+    - name: REPO_OWNER
+      description: "GitHub org that triggered the prowjob. Variable set by prow. Prow will not set this variable if the prowjob type is periodic. In that case an empty string default value will be used."
+      type: string
+      default: ""
+    - name: REPO_NAME
+      description: "GitHub repo that triggered the prowjob. Variable set by prow. Prow will not set this variable if the prowjob type is periodic. In that case an empty string default value will be used."
```
Comment (IwonaLanger) [inline task/go-unit-tests/0.1/go-unit-tests.yaml:31]:
```suggestion
      description: "GitHub repo that triggered the ProwJob. A variable set by Prow. Prow does not set this variable if the ProwJob type is periodic. In that case, an empty string default value is used."
```

File: task/go-unit-tests/0.1/samples/prowjob-testing-go-code.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+presubmits:
+  kyma-project/test-infra:
+    # Prow set pipeline params according to the list of job environment variables.
+    # See https://docs.prow.k8s.io/docs/jobs/#job-environment-variables
+    # The pipeline can use these environment variables to get the refs and extra-refs to clone.
+    - name: tekton-demo
+      # Default Prow decoration must be disabled because the pipeline uses its own decoration.
+      decorate: false
+      # Prowjobs using Tekton pipelines must use the tekton-pipeline agent.
```
Comment (IwonaLanger) [inline task/go-unit-tests/0.1/samples/prowjob-testing-go-code.yaml:9]:
```suggestion
      # ProwJobs using Tekton pipelines must use the tekton-pipeline agent.
```

File: task/go-unit-tests/0.1/samples/prowjob-testing-go-code.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+              # Prow set pipeline params according to the list of job environment variables.
+              - name: JOB_SPEC
+                type: string
+              - name: REPO_OWNER
+                type: string
+              - name: REPO_NAME
+                type: string
+            description: "Tekton Demo"
+            tasks:
+              # clone-refs must run first to clone repositories refs provided by prowjob.
```
Comment (IwonaLanger) [inline task/go-unit-tests/0.1/samples/prowjob-testing-go-code.yaml:44]:
```suggestion
              # clone-refs must run first to clone repositories refs provided by ProwJob.
```

**Review by Ressetkk — DISMISSED** [review]

File: task/go-unit-tests/0.1/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+## Workspaces
+
+- **repo**: The workspace where tested sources are stored. Usually, this should be a workspace shared with other
+  tasks. _(REQUIRED)_
+
+## Parameters
+
+- **path-to-test**: The path to tested Go code. _(OPTIONAL, default: "./...")_
+- **REPO_OWNER**: GitHub org that triggers the ProwJob. A variable set by Prow. Prow does not set this variable if the
```
Comment (IwonaLanger) [inline task/go-unit-tests/0.1/README.md:25]:
```suggestion
- **REPO_OWNER**: GitHub organization that triggers the ProwJob. A variable set by Prow. Prow does not set this variable if the
```

File: task/go-unit-tests/0.1/README.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  ProwJob type is periodic. In that case, an empty string default value is used. _(OPTIONAL, default: "")_
+- **REPO_NAME**: GitHub repo that triggers the ProwJob. A variable set by Prow. Prow does not set this variable if the
```
Comment (IwonaLanger) [inline task/go-unit-tests/0.1/README.md:27]:
```suggestion
- **REPO_NAME**: GitHub repository that triggers the ProwJob. A variable set by Prow. Prow does not set this variable if the
```

File: task/go-unit-tests/0.1/go-unit-tests.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    - name: repo
+      description: "The workspace where tested sources are stored. Usually, this should be a workspace shared with other tasks."
+      mountPath: /home/prow/go
+  params:
+    - name: path-to-test
+      description: "The path to tested Go code."
+      type: string
+      default: "./..."
+    - name: REPO_OWNER
+      description: "GitHub org that triggers the ProwJob. A variable set by Prow. Prow does not set this variable if the ProwJob type is periodic. In that case, an empty string default value is used."
```
Comment (IwonaLanger) [inline task/go-unit-tests/0.1/go-unit-tests.yaml:28]:
```suggestion
      description: "GitHub organization that triggers the ProwJob. A variable set by Prow. Prow does not set this variable if the ProwJob type is periodic. In that case, an empty string default value is used."
```

File: task/go-unit-tests/0.1/go-unit-tests.yaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+      type: string
+      default: ""
+    - name: REPO_NAME
+      description: "GitHub repo that triggered the ProwJob. A variable set by Prow. Prow does not set this variable if the ProwJob type is periodic. In that case, an empty string default value is used."
```
Comment (IwonaLanger) [inline task/go-unit-tests/0.1/go-unit-tests.yaml:32]:
```suggestion
      description: "GitHub repository that triggers the ProwJob. A variable set by Prow. Prow does not set this variable if the ProwJob type is periodic. In that case, an empty string default value is used."
```

File: task/go-unit-tests/0.1/README.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
+# Unit Tests for Go Code
+
+A go-unit-tests Task is used to run unit tests for Go code.
+The task is tailored to be triggered by Prow as a ProwJob.
+It uses a built-in go test command to run unit tests.
```
[the remark this answers is not in the document]
Comment (dekiel) [reply task/go-unit-tests/0.1/README.md:5]: The full command is `go test`

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**